-----------------------------
-- GameSections --
-----------------------------
-----------------------------
-- SERVICES --
-----------------------------

-----------------------------
-- VARIABLES --
-----------------------------
local GameSections = {}

-----------------------------
-- PUBLIC VARIABLES --
-----------------------------
GameSections.Objectives = {
	["Money"] = { Text = "Earn $500 to pay for rent." },
	["Enter"] = { Text = "Enter the Krusy Krab", Fulfilled = false },
}

-----------------------------
-- PRIVATE FUNCTIONS --
-----------------------------

-----------------------------
-- PUBLIC FUNCTIONS --
-----------------------------

-----------------------------
-- MAIN --
-----------------------------
return GameSections
